<launch>

   <arg name="use_gazebo" default="false" />
   <arg name="map_name" default="maze.yaml"/>

   <!-- Localization -->
   <include file="$(find enawpl)/launch/localization.launch">
      <arg name="use_gazebo" value="$(arg use_gazebo)" />
      <arg name="map_name" default="$(arg map_name)"/>
   </include>

   <!-- Path planning -->
   <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <param name="base_global_planner" value="enawpl/NavfnROS" />
      <rosparam file="$(find enawpl)/config/move_base.yaml" command="load" />
      <rosparam file="$(find enawpl)/config/costmap_common.yaml" command="load" ns="global_costmap" /> 
      <rosparam file="$(find enawpl)/config/costmap_common.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find enawpl)/config/costmap_global.yaml" command="load" /> 
      <rosparam file="$(find enawpl)/config/costmap_local.yaml" command="load" />
      <rosparam file="$(find enawpl)/config/planner_global.yaml" command="load" />
      <rosparam file="$(find enawpl)/config/planner_local.yaml" command="load" />
      <rosparam file="$(find enawpl)/config/energy_aware_planner_params.yaml" command="load" />
   </node>

   <node pkg="enawpl" type="zone_visualizer.py" name="zone_visualizer" output="screen">
      <param name="yaml_path" value="$(find enawpl)/config/energy_aware_planner_params.yaml" />
   </node>

   <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tutorial_pkg)/rviz/move_base.rviz"/>

</launch> 
